BoxContentSDK/BoxContentSDK/Models/BOXMetadataUpdateTask.m

```diff
@@ -0,0 +1,58 @@
+//
+//  BOXMetadataUpdate.m
+//  BoxContentSDK
+//
+//  Created on 6/15/15.
+//  Copyright (c) 2015 Box. All rights reserved.
+//
+
+#import "BOXMetadataUpdateTask.h"
+#import "BOXLog.h"
+
+@implementation BOXMetadataUpdateTask
+
+- (instancetype)initWithOperation:(BOXMetadataUpdateOperation)operation path:(NSString *)path value:(NSString *)value
+{
+    if (self = [self init]) {
+        NSString *formatString = nil;
+        if (path) {
+            formatString = @"/%@";
+            if ([path characterAtIndex:0] == '/') {
+                formatString = @"%@";
+            }
+        }
+        
+        self.value = value;
+        self.path = [NSString stringWithFormat:formatString, path];
+        self.operation = operation;
+        
+        [self validate];
+    }
+    
+    return self;
+}
+
+- (void)validate
+{
+    BOXAssert([self BOXMetadataUpdateOperationToString:self.operation], @"The operation parameter for BOXMetadataUpdateTask cannot be nil. Please make sure to use an ENUM within BOXMetadataUpdateOperation.");
+    BOXAssert(self.path, @"The path parameter for BOXMetadataUpdateTask cannot be nil");
+    BOXAssert(!(self.operation == BOXMetadataUpdateADD || self.operation == BOXMetadataUpdateREPLACE) || self.value, @"The value parameter cannot be nil when using BOXMetadataUpdateADD or BOXMetadataUpdateREPLACE.");
+}
+
+- (NSString *)BOXMetadataUpdateOperationToString:(BOXMetadataUpdateOperation)operation
```

Comment:
comment in here somewhere about how these are not meant to be localized (functional, not user-facing)